pkg/event/event_unmarshal.go

```diff
@@ -152,15 +155,29 @@ func readDataValue(iter *jsoniter.Iterator) ([]DataValue, error) {
 			case "valueType":
 				dv.ValueType = ValueType(iter.ReadString())
 			case "value":
-				cacheValue = iter.ReadString()
+				cacheValue = iter.Read()
 			default:
 				iter.Skip()
 			}
 		}
+
 		if dv.ValueType == DECIMAL {
-			dv.Value, err = strconv.ParseFloat(cacheValue, 64)
+			dv.Value, err = strconv.ParseFloat(cacheValue.(string), 64)
+		} else if dv.ValueType == ENUMERATION {
+			dv.Value = cacheValue.(string)
+		} else if dv.ValueType == REDFISH_EVENT {
+			jsonRedfishEvent, err := json.Marshal(cacheValue)
+			if err != nil {
+				return values, err
+			}
+
+			e := redfish.Event{}
+			if err := json.Unmarshal(jsonRedfishEvent, &e); err != nil {
+				return values, err
+			}
+			dv.Value = e
 		} else {
-			dv.Value = SyncState(cacheValue)
+			return values, fmt.Errorf("value type %v is not supported", dv.ValueType)
```

Comment:
we have to make sure the test passes here. Also in this case we may want to throw an exception while marshaling the event object?

Reply:
updated to throw an exception and added test case "struct Data unSupportedValueType" in event_unmarshal_test.go.

Reply:
done